iceoryx_hoofs/include/iceoryx_hoofs/cxx/expected.hpp

```diff
@@ -123,6 +124,12 @@ struct error
     template <typename... Targs>
     error(Targs&&... args) noexcept;
 
+    template <typename F>
+    static error<T> from(F e)
+    {
+        return error(cxx::from<F, T>(e));
+    }
+
```

Comment:
I'll remove this from the current PR since it is not related to the issue but I would like to introduce this as feature in another PR. Any objections?

Reply:
@elBoberido For me, it's ok to introduce it as part of this PR.

Reply:
Well, I would have to create tests and add documentation. For traceability it would also not quite fit to this issue

Reply:
Why not add a link to the follow-up issue here and maybe also as a `/// @ todo` in code? That workflow is ok for me as long as one continues to work on the follow-up in the next few weeks.

Reply:
done with #1012